### PR TITLE
AP_Scripting: Remove send_text if directory is created

### PR DESCRIPTION
**Pull request by rishabsingh3003**
As per the discussion in https://github.com/ArduPilot/ardupilot/pull/14123#issuecomment-646983985
Tested in SITL
@peterbarker @rmackay9

### CODE REVIEW

**Comment by rishabsingh3003** [comment]
Travis has failed for some unrelated reason.

**Comment by peterbarker** [comment]
It might amuse you to know that the first SD card I tried this on I'd get the "directory created" every time I rebooted.

The SD card is terminally corrupt; it's an image that @WickedShell sent me.

So at least in that case it was a hint that storage was bad.

File: libraries/AP_Scripting/AP_Scripting.cpp [hunk 1]
```diff
@@ -97,12 +97,13 @@ void AP_Scripting::init(void) {
     }
 
     const char *dir_name = SCRIPTING_DIRECTORY;
-
-    if (!AP::FS().mkdir(dir_name)) {
-        gcs().send_text(MAV_SEVERITY_INFO, "Lua: created new directory (%s)", dir_name);
-    } else if (errno != EEXIST) {
-        gcs().send_text(MAV_SEVERITY_INFO, "Lua: failed to create (%s)", dir_name);
-        return;
+    const bool directory_created = !AP::FS().mkdir(dir_name);
```
Comment (peterbarker) [inline libraries/AP_Scripting/AP_Scripting.cpp:100]: Please remove this temp.  It's just polluting the namespace of the rest of the function.

File: libraries/AP_Scripting/AP_Scripting.cpp [hunk 2]
```diff
@@ -97,12 +97,13 @@ void AP_Scripting::init(void) {
...
+    if (!directory_created) {
+        if (errno != EEXIST) {
+            //alert GCS that directory could not be found/made, despite lua script being enabled
```
Comment (peterbarker) [inline libraries/AP_Scripting/AP_Scripting.cpp:103]: Comment is confusing and kind of redundant :-)

**Comment by WickedShell** [comment]
> The SD card is terminally corrupt; it's an image that @WickedShell sent me.

If you remount it with the PR from tridge it will work...

File: libraries/AP_Scripting/AP_Scripting.cpp [hunk 3]
```diff
@@ -97,12 +97,11 @@ void AP_Scripting::init(void) {
     }
 
     const char *dir_name = SCRIPTING_DIRECTORY;
-
-    if (!AP::FS().mkdir(dir_name)) {
-        gcs().send_text(MAV_SEVERITY_INFO, "Lua: created new directory (%s)", dir_name);
-    } else if (errno != EEXIST) {
-        gcs().send_text(MAV_SEVERITY_INFO, "Lua: failed to create (%s)", dir_name);
-        return;
+    if (AP::FS().mkdir(dir_name)) {
```
Comment (peterbarker) [inline libraries/AP_Scripting/AP_Scripting.cpp:100]: Did you lose the `!` here?

Comment (rmackay9) [reply libraries/AP_Scripting/AP_Scripting.cpp:100]: @peterbarker, I think it's correct as it is... either that or it was always printing it backwards.

Comment (peterbarker) [reply libraries/AP_Scripting/AP_Scripting.cpp:100]: Yep.  Thought mkdir may have gone bool as part of the abstraction.

Comment (peterbarker) [reply libraries/AP_Scripting/AP_Scripting.cpp:100]: ... it should probably say `!=0`, it occurs to me, to make things clearer.

**Comment by peterbarker** [comment]
Merged, thanks!

File: libraries/AP_Scripting/AP_Scripting.cpp (hunk 3 above)
Comment (rmackay9) [reply libraries/AP_Scripting/AP_Scripting.cpp:100]: @peterbarker, it's super weird that mkdir returns 0 if it works.  I guess that's a linux-y thing..

**Pull request merged**